core-bundle/src/EventListener/DataContainer/DefaultOperationsListener.php

```diff
@@ -106,7 +106,7 @@ private function getDefaults(string $table): array
             if (DataContainer::MODE_TREE_EXTENDED !== ($GLOBALS['TL_DCA'][$ctable]['list']['sorting']['mode'] ?? null)) {
                 $operations += [
                     'children' => [
-                        'href' => 'table='.$ctable,
+                        'href' => 'table='.$ctable.($ctable === $table ? '&amp;ptable='.$table : ''),
```

Comment:
```suggestion
                        'href' => 'table='.$ctable.($hasPtable && $ctable === $table ? '&amp;ptable='.$table : ''),
```

Don‘t we have to check `$hasPtable` here as well?

Reply:
Not sure. This is the children operation, so in theory it is independent of the `ptable`. But if the current table and the **child table** is the same, we need to add `ptable=...` to the URL. That currently only applies to nested elements, but I guess you know that 😆

Reply:
I ran some more tests and you are right. Using `$hasPtable` would be wrong here.